demos/ret2spec_common.cc

```diff
@@ -0,0 +1,101 @@
+#include <array>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "cache_sidechannel.h"
+#include "instr.h"
+#include "ret2spec_common.h"
+#include "utils.h"
+
+// We cannot include the private_data twice, because that would lead to linking
+// error. Declaring it extern in here.
+extern const char *private_data;
+
+// Calls sched_yield in the cross-address-space version.
+void (*conditionally_unschedule)();
+// Calls sched_yield in the cross-address-space version, in the
+// same-address-space version starts the ReturnsFalse recursion.
+void (*unschedule_or_start_returns_false)();
```

Comment:
Suggest making these static variables (or putting them in an empty namespace) and exporting a function to set them.

Reply:
This cannot be made static because it's used cross-file.